src/xdp-utils.c

```diff
@@ -591,19 +548,27 @@ xdp_get_app_info_from_pid (pid_t pid,
   g_autoptr(XdpAppInfo) app_info = NULL;
   g_autoptr(GError) local_error = NULL;
 
-  /* TODO: Handle snap support via apparmor here */
-
   app_info = parse_app_info_from_flatpak_info (pid, &local_error);
   if (app_info == NULL && local_error)
     {
       g_propagate_error (error, g_steal_pointer (&local_error));
       return NULL;
     }
 
+  if (app_info == NULL)
+    {
+      app_info = parse_app_info_from_snap (pid, &local_error);
+      if (app_info == NULL && local_error)
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return NULL;
+        }
+    }
+
   if (app_info == NULL)
     app_info = xdp_app_info_new_host ();
 
-  return app_info;
+  return g_steal_pointer (&app_info);
```

Comment:
How did this even work before?

Reply:
I don't think it ever did.  Previously it was only used by the `Device.AccessDevice()` portal API.  As I understand it, that API was called by an abandoned set of Pulse Audio patches.  It looks like it was being called by a Pipewire module in the past too, but that seems to have been removed.

Presumably it worked at some point in the past but was cleaned up without proper testing.